archive/src/main/java/io/orkes/conductor/dao/postgres/archive/PostgresArchiveDAO.java

```diff
@@ -536,4 +540,124 @@ public List<WorkflowModel> getWorkflowFamily(String workflowId, boolean summaryO
             throw new RuntimeException(e);
         }
     }
+
+    private String[] getLabels(String description) {
+        if (description == null || description.isEmpty()) {
+            return new String[0];
+        }
+        try {
+            JsonNode jsonDescription = objectMapper.readTree(description);
+            JsonNode labels = jsonDescription.get("labels");
+
+            if (labels != null && labels.isArray()) {
+                String[] labelsArray = new String[labels.size()];
+                for (int i = 0; i < labels.size(); i++) {
+                    labelsArray[i] = labels.get(i).asText();
+                }
+                return labelsArray;
+            }
+        } catch (JsonProcessingException e) {
+            log.error("Unable to get labels from {}", description, e);
+
+        }
+        return new String[0];
+    }
+
+    @Override
+    public List<String> getUserWorkflowIds(List<String> groupsAndRoles) {
+        final String GET_USER_WORKFLOWS = "SELECT " +
+                "workflow_id FROM archive.workflow_archive " +
+                "WHERE rbac_labels && ARRAY[?]::text[]";
```

Comment:
If you use ';' in the end of SQL command in another query then add it also here.

Reply:
added